### PR TITLE
Fix flaky e2e-tests-dashboard-ee

### LINKED ISSUE #28491: Fix e2e-tests-dashboard-ee
Reference: [Slack convo (internal)](https://metaboat.slack.com/archives/C5XHN8GLW/p1676997906308539)

This issue is specifically for fixing `frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js`

If you pause Cypress before it clicks the info button, just before this line:
https://github.com/metabase/metabase/blob/c82ccb1d47c74fd036e6871bbfbe87349ce7efdc/frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js#L72

you'll see that there's still 2 more API endpoints being responded

This means it could be a race condition (I didn't take much time into troubleshooting this yet, but this is all based on my observation):
1. Cypress click the info button revealing the right sidebar
2. API responses received, updating the UI prompting React to rerender the UI, and now the right side bar is close.
3. Cypress don't see the right sidebar so the test failed.

### PR DESCRIPTION
**Pull request by WiNloSt**
> **Note**
> Might be superseded by https://github.com/metabase/metabase/pull/28505

Closes https://github.com/metabase/metabase/issues/28491
### Example failures
- https://www.deploysentinel.com/ci/runs/63f4f8d52532afe2734d86bc
- https://www.deploysentinel.com/ci/runs/63f4d6ee6173177b78b8c5bb
- https://www.deploysentinel.com/ci/runs/63f4c7f8b7c78651abafa3f4

### Description

This fix solves the problem by waiting for the timestamp label to update to make sure that the UI won't rerender after we click the info button, otherwise, it could close the right sidebar unintentionally. This issue is a race condition.

### How to verify

Run E2E tests in `frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js` multiple times, and they shouldn't fail.

### CODE REVIEW

File: frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ describe("scenarios > dashboard", () => {
       .type("{selectall}Orders per year")
       .blur();
 
-    cy.wait("@updateDashboard");
+    cy.findByText("Edited a few seconds ago by you").should("be.visible");
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js:70]: There seem to be other 2 API responses after waiting for `@updateDashboard` previously as seen in this screenshot.

image

This fix makes sure we wait for those API calls to finish before clicking the info button.

Comment (npfitz) [reply frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js:70]: Does seem like a better thing to wait for haha. Thanks @WiNloSt

**Review by npfitz — APPROVED** [review]

File: frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js [hunk 2]
```diff
@@ -67,7 +67,7 @@ describe("scenarios > dashboard", () => {
       .type("{selectall}Orders per year")
       .blur();
 
-    cy.wait("@updateDashboard");
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/dashboard/dashboard.cy.spec.js:70]: If you are removing the wait, remove the `intercept` as well.

Btw, nothing is stopping you from waiting for this request to come back AND to assert on a string in the UI after that.

**Comment by WiNloSt** [comment]
closing in favor of https://github.com/metabase/metabase/pull/28505

**Pull request closed without merging**